<?xml version="1.0"?>

<launch>
  <arg name="standalone" default="false" doc="if this is set to the string true, does not spawn a nodelet manager. Otherwise it does."/>

  <group if="$(arg standalone)">
      <node pkg="nodelet" type="nodelet" name="bluerov_nodelet_manager" args="manager" output="screen" respawn="true">
          <param name="num_worker_threads" value="3" />
      </node>
  </group>

  <node pkg="nodelet" type="nodelet" name="stag_ros" output="screen" args="load stag_ros/StagNodelet bluerov_nodelet_manager" respawn="false" required="true">
    <param name="camera_image_topic" type="string" value="/camera_array/cam0/image_raw" />
    <param name="camera_info_topic" type="string" value="/camera_array/cam0/camera_info" />
    <param name="tag_id_type" type="int" value="21" />
    <param name="marker_frame_prefix" type="string" value="/ar_marker_" />
    <param name="default_marker_size" type="double" value="0.100" />
    <param name="output_frame_id"     type="string" value="/base_link" />
    <param name="image_frame_id"      type="string" value="/cam_0_optical_frame" />
    <param name="marker_message_topic" type="string" value="/bluerov_controller/ar_tag_detector" />
    <param name="process_images_in_parallel" type="bool" value="true" />
    <param name="use_marker_bundles" type="bool" value="true" />
    <param name="publish_debug_images" type="bool" value="false" />
    <rosparam command="load" file="$(find droplet_underwater_assembly)/param/bundle-four-stags-aluminum.json" />

    <rosparam command="load" param="marker_sizes_by_id" file="$(find droplet_underwater_assembly)/param/marker_sizes_by_id.yaml" />

    <param name="track_markers" type="bool" value="true" />
    <param name="marker_track_height_offset" type="int" value="50" />
    <param name="marker_track_width_offset" type="int" value="50" />
  </node>
</launch>
